### DISCUSSION TITLE
Bug: lineStyle alignment 1 option for Graphics set to inner depending on start/end position

### DISCUSSION THREAD

**Discussion by whitespacecode** [pixijs/pixijs#9149] [Q&A] [answered]
### Current Behavior

When drawing lines and and using the alignment option 1 = Outer

> The alignment of any lines drawn (0.5 = middle, 1 = outer, 0 = inner). 
> Default Value: 0.5
 
Depending on where the line start and moves to, the 'outer' option suddenly becomes 'inner'

### Expected Behavior

The alignment 1 should always be 'outer'

### Steps to Reproduce

I have created a fiddle with a trimmed downed version of what i'm using it for.
You can draw lines to create a room. And the inside the room we drag objects around the are depending on the lines to have outer.

### Environment

- **`pixi.js` version**: *e.g. 6.5.4*
- **Browser & Version**: *e.g. Chrome 108*
- **OS & Version**: *e.g. macos*
- **Running Example**: *e.g. <https://jsfiddle.net/joerithegreat/82Lb70k4/7/>*

**Comment by bigtimebuddy** [comment, marked as answer]
Graphics line alignment is determined by the orientation of the points (clockwise vs counter-clockwise). Here's a simplified version of your example: https://jsfiddle.net/bigtimebuddy/c5w9nefq/

It doesn't have anything to do with the position of the first point.

For example, this shape for example is counter-clockwise in the direction of the points. This will be the inverse for the alignment (inner = 1, outer = 0). 
```json
[
  {"x": 0,"y": 100},
  {"x": 200,"y": 100},
  {"x": 200,"y": 0},
  {"x": 0,"y": 0},
]
```

If this happens you can reverse the points:

```js
[
  {"x": 0,"y": 100},
  {"x": 200,"y": 100},
  {"x": 200,"y": 0},
  {"x": 0,"y": 0},
].reverse()
```

**Reply by whitespacecode** [reply]
Reversing was not really an option. What is did is creating a function that determine the direction of my room walls.
```
isClockwise(walls) {
  let sum = 0;
  for (let i = 0; i < walls.length; i++) {
     let j = (i + 1) % walls.length;
         sum += (walls[j].from.x - walls[i].from.x) * (walls[j].from.y + walls[i].to.y);
    }
  return sum > 0;
}
```
I then use the `isClockwise` to set the alignment either 1 or 0.
Thank you for the insight!

**Comment by whitespacecode** [comment]
> If this happens you can reverse the points

Ooh i didn't know that, but what if you don't know the orientation?
Is there something you can determine the orientation of the points with?

**Comment by bigtimebuddy** [comment]
There a few 3rd-party utils you can use for testing: https://www.npmjs.com/search?q=polygon%20ccw
